configuration/postgres/postgres.go

```diff
@@ -0,0 +1,421 @@
+		}
+	}()
+	payload := make(map[string]interface{})
+	err := json.Unmarshal([]byte(msg.Payload), &payload)
+	if err != nil {
+		p.logger.Errorf("error in unmarshal: ", err)
+		return
+	}
+	var key, value, version string
+	m := make(map[string]string)
+	// trigger should encapsulate the row in "data" field in the notification
+	row := reflect.ValueOf(payload[payloadDataKey])
+	if row.Kind() == reflect.Map {
+		for _, k := range row.MapKeys() {
+			v := row.MapIndex(k)
+			strKey := k.Interface().(string)
+			switch strings.ToLower(strKey) {
+			case "key":
+				key = v.Interface().(string)
+				if yes := p.isSubscribed(subscriptionID, channel, key); !yes {
+					p.logger.Debugf("ignoring notification for %v", key)
+					return
+				}
+			case "value":
+				value = v.Interface().(string)
+			case "version":
+				version = v.Interface().(string)
+			case "metadata":
+				a := v.Interface().(map[string]interface{})
+				for k, v := range a {
+					m[k] = v.(string)
+				}
+			}
+		}
+		e := &configuration.UpdateEvent{
+			Items: map[string]*configuration.Item{
+				key: {
+					Value:    value,
+					Version:  version,
+					Metadata: m,
+				},
+			},
+			ID: subscriptionID,
+		}
+		err = handler(ctx, e)
+		if err != nil {
+			p.logger.Errorf("fail to call handler to notify event for configuration update subscribe: %s", err)
+		}
+	} else {
+		p.logger.Info("unknown format of data received in notify event - '%s'", msg.Payload)
+	}
+}
+
+func parseMetadata(cmetadata configuration.Metadata) (metadata, error) {
+	m := metadata{}
+	if val, ok := cmetadata.Properties[connectionStringKey]; ok && val != "" {
+		m.connectionString = val
+	} else {
+		return m, fmt.Errorf(ErrorMissingConnectionString)
+	}
+	if tbl, ok := cmetadata.Properties[configtablekey]; ok && tbl != "" {
+		if !allowedChars.MatchString(tbl) {
+			return m, fmt.Errorf("invalid table name : '%v'. non-alphanumerics are not supported", tbl)
+		}
+		if len(tbl) > maxIdentifierLength {
+			return m, fmt.Errorf(ErrorTooLongFieldLength+" - tableName : '%v'. max allowed field length is %v ", tbl, maxIdentifierLength)
+		}
+		m.configTable = tbl
+	} else {
+		return m, fmt.Errorf(ErrorMissingTableName)
+	}
+	// configure maxTimeout if provided
+	if mxTimeout, ok := cmetadata.Properties[connMaxIdleTimeKey]; ok && mxTimeout != "" {
+		if t, err := time.ParseDuration(mxTimeout); err == nil && t > 0 {
+			m.maxIdleTimeout = t
+		}
+	}
+	if m.maxIdleTimeout == 0 {
+		m.maxIdleTimeout = defaultMaxConnIdleTime
+	}
+	return m, nil
+}
+
+func Connect(ctx context.Context, conn string, maxTimeout time.Duration) (*pgxpool.Pool, error) {
+	config, err := pgxpool.ParseConfig(conn)
+	if err != nil {
+		return nil, fmt.Errorf("postgres configuration store connection error : %s", err)
+	}
+	pool, err := pgxpool.NewWithConfig(ctx, config)
+	if err != nil {
+		return nil, fmt.Errorf("postgres configuration store connection error : %s", err)
+	}
+	pingErr := pool.Ping(ctx)
```

Comment:
probably not right now ... but consider implementing the Pinger interface for this component ....

Reply:
@mukundansundar - please add this to the issue tracking configstore improvements - https://github.com/dapr/dapr/issues/5183